shared/tests/utils/test_git_util.py

```
@@ -191,3 +191,253 @@ def test_clone_repo_with_token_failure(self, mock_subprocess):
+    def test_clone_repo_with_token_gerrit_at_symbol(
+        self, mock_setup_hooks, mock_subprocess
+    ):
+        """Test that Gerrit tokens with @ symbol are properly URL encoded"""
+        mock_result = MagicMock()
+        mock_result.returncode = 0
+        mock_subprocess.return_value = mock_result
+        mock_setup_hooks.return_value = (True, None)
+
+        project_url = "https://gerrit.example.com/project"
+        branch = "main"
+        project_path = "/tmp/test-repo"
```

Ruff (0.14.11)

Probable insecure usage of temporary file or directory: "/tmp/test-repo"

(S108)